ha/core/controller/pcs/cluster_controller.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+from ha.core.error import ValidationFailedError
+from ha.core.controller.element_controller import ElementController
+from ha.core.error import HAUnimplemented
+
+class PcsClusterController(ElementController):
+    """ Pcs cluster controller to perform pcs cluster level operation. """
+    def __init__(self):
+        """
+        Initalize pcs cluster controller
+        """
+        super(PcsClusterController, self).__init__()
+        self._actions: list =  [method for method in dir(PcsClusterController)
+            if method.startswith("_") is False]
+
+    def _validate(self, args: dict) -> None:
```

Comment:
shall we name it _validate_actions?
OR _validate can be used for other validations too? Probably there are no other validations.

Reply:
It is validating actions and args for action if needed